src/realm/object-store/shared_realm.cpp

```diff
@@ -406,9 +406,12 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
     Schema actual_schema = get_full_schema();
 
     // Frozen Realms never modify the schema on disk and we just need to verify
-    // that the requested schema is a subset of what actually exists
+    // that the requested schema is compatible with what actually exists on disk
+    // at that frozen version. Adding new tables is allowed as those can be backed
+    // by empty Results, breaking changes are not allowed.
```

Comment:
```suggestion
    // at that frozen version. Tables are allowed to be missing as those can be
    // represented by empty Results, but tables which exist must have all of the
    // requested properties with the correct type.
```

I think "Adding new tables" is potentially confusing here.